packages/feedback/components/feedback/src/feedback.ts

```diff
@@ -0,0 +1,381 @@
+/**
+ * @license
+ *
+ * Copyright IBM Corp. 2023
+ *
+ * This source code is licensed under the Apache-2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import { LitElement, TemplateResult, html } from 'lit';
+import { property, state } from 'lit/decorators.js';
+import { v4 as uuidv4 } from 'uuid';
+import HostListener from '@carbon/web-components/es/globals/decorators/host-listener';
+import HostListenerMixin from '@carbon/web-components/es/globals/mixins/host-listener';
+
+// @ts-ignore
+import styles from './feedback.scss?inline';
+import { FeedbackData } from '../defs';
+
+/**
+ * Feedback component to record and give feedback on AI generated content
+ */
+export class Feedback extends HostListenerMixin(LitElement) {
+  static styles = styles;
+
+  /**
+   * Model ID or Model Name for which feedback is recording
+   */
+  @property({ attribute: 'model', type: String })
+  private _model_id = '';
+
+  /**
+   * User Input to the model
+   */
+  @property({ attribute: 'input', type: String })
+  private _input = '';
+
+  /**
+   * Output generated by AI Model
+   */
+  @property({ attribute: 'output', type: String })
+  private _output = '';
+
+  /**
+   * ID generated For a particular input and output
+   */
+  @property({ attribute: 'generation-id', type: String, reflect: true })
+  generation_id;
+
+  @property()
+  private feedbackList: FeedbackData[] = [];
+  /**
+   * State variable for Feedback Model
+   */
+  @state()
+  private isModelOpen = false;
+
+  /**
+   * State variable for Edit selected text
+   */
+  @state()
+  private isEditable = false;
+
+  /**
+   * To store data of selected text
+   */
+  @state()
+  private selection;
+
+  @state()
+  private disableSave = false;
+
+  @state()
+  private isUpdateMode = false;
+
+  @state()
+  highlighted: TemplateResult | null = null;
+
+  /**
+   * Getter for selection
+   */
+  get Selection() {
+    return this.selection;
+  }
+
+  /**
+   * Object for recording the feedback
+   */
+  @state()
+  private formData: FeedbackData = {
+    feedback_id: '',
+    generation_id: '',
+    start_index: 0,
+    end_index: 0,
+    selected_text: '',
+    suggested_text: '',
+    feedback_type: [],
```

Comment:
Can we stick with camelCasing of variables?

Reply:
Done